bindings/java/src/org/sleuthkit/datamodel/timeline/ArtifactEventType.java

```diff
@@ -50,64 +42,56 @@
 	BlackboardAttribute.Type getDateTimeAttributeType();
 
 	/**
-	 * Get the ID of the the artifact type that this EventType is derived from.
-	 *
-	 * @return the ID of the the artifact type that this EventType is derived
-	 *         from.
+	 * @return a function from an artifact to a String to use as part of the
+	 *         full event description
 	 */
-	default int getArtifactTypeID() {
-		return getArtifactType().getTypeID();
-	}
+	String extractFullDescription(BlackboardArtifact artf) throws TskCoreException;
```

Comment:
Do APIs such as this belong as pubic APIs at the SleuthKit level? They seem really specific to the UI, a criticism I had for many of the CommunicationsManager APIs. I am also a bit confused by the use case, given that we are passing artifacts into these methods, where it seems that the artifact would be part of the state of the event. I would like to have some time to discuss where all this is going; perhaps I am hampered by limited familiarity with the Autopsy timeline code.

By the way, artf seems like an awkward and unpronounceable abbreviation. Why not just spell out artifact?